src/hooks/posters.tsx

```diff
@@ -162,5 +182,31 @@ export const usePosters = (posterHallId: string) => {
     setSearchInputValue,
     setLiveFilter,
     setBookmarkedFilter,
-  };
+  } as PostersContextState;
+};
+
+export const PostersContext = createContext<PostersContextState | undefined>(
+  undefined
+);
+
+type PostersProviderProps = Readonly<{ venueId: string }>;
+export const PostersProvider: React.FC<PostersProviderProps> = ({
+  venueId,
+  children,
+}) => (
+  <PostersContext.Provider value={usePosters(venueId)}>
```

Comment:
I don't know that it's even valid to use hooks like this? At the very least, we shouldn't be. Please call the hook outside of the JSX, set the value to a `const`, then pass that through to the `value` prop.